### PR TITLE
Align Travis configuration with AppVeyor in #86

### PR DESCRIPTION
**Pull request by cirulls**
## Summary
This pull request aligns the Travis configuration with the AppVeyor configuration implemented in #86.

## Why
#76 introduced a test for checking that the XProc harness is working. Running XProc requires XMLCalabash which is installed in the ```.travis.yml``` file. However, XMLCalabash currently comes bundled with Saxon 9.7.0.11 as highlighted by @AirQuick in https://github.com/xspec/xspec/pull/86#discussion_r104289634. It is therefore useless to install it and run the test for XProc when a lower version of Saxon is used. This also makes the test suite slower.    

## What has changed
In ```.travis.yml``` I added the ```XMLCALABASH_VERSION``` in the matrix as suggested by @AirQuick in https://github.com/xspec/xspec/pull/86#discussion_r104293014. This is in line with the configuration in ```appveyor.yml``` and also makes it easier to upgrade to a new version of XMLCalabash in the future.

In ```xspec.bats``` I skipped the test if the environment variable ```XMLCALABASH_VERSION``` is not set.   

## How to review
I tested that the Travis configuration works as expected in [this Travis build](https://travis-ci.org/xspec/xspec/builds/208357031) (see the job logs):

- [build 235.1](https://travis-ci.org/xspec/xspec/jobs/208357032) (Saxon 9.7.0-15) installs XMLCalabash and runs the test
- [build 235.2](https://travis-ci.org/xspec/xspec/jobs/208357033) (Saxon 9.6.0-10) doesn't install XMLCalabash and skip the test
- [build 235.3](https://travis-ci.org/xspec/xspec/jobs/208357035) (Saxon 9.6.0-7) doesn't install XMLCalabash and skip the test

@AirQuick : could you please review this pull request and see if it aligns with your pull request in #86? Once you approve this, I will merge both pull requests into master.

### CODE REVIEW

File: test/xspec.bats [hunk 1]
```diff
@@ -193,6 +193,10 @@ teardown() {
 
 
 @test "executing the Saxon XProc harness generates a report with UTF-8 encoding" {
+	if [ -z ${XMLCALABASH_VERSION} ]; then
```
Comment (AirQuick) [inline test/xspec.bats]: With this line, to run `xspec.bats` thoroughly, you have to set both `XMLCALABASH_VERSION` and `XMLCALABASH_CP`.
Can you consider checking the existence of `XMLCALABASH_CP` instead of `XMLCALABASH_VERSION`? That makes it a little bit easier to run `xspec.bats` on local console without `.travis.yml`, which I do occasionally.

Comment (cirulls) [inline test/xspec.bats]: The check in ```xspec.bats``` is now based on XMLCALABASH_CP instead of XMLCALABASH_VERSION (which makes more sense because XMLCalabash is properly installed only once XMLCALABASH_CP is set). 

Have you got any further changes?

**Comment by cirulls** [comment]
Thanks for reviewing the pull request and suggest changes. This is good to go and I'll merge into master.

File: test/xspec.bats [hunk 2]
```diff
@@ -145,56 +145,60 @@ teardown() {
...
-	run java -Xmx1024m -cp ${XMLCALABASH_CP} com.xmlcalabash.drivers.Main -isource=xspec-72.xspec xspec-home=file:${PWD}/../ -oresult=xspec/xspec-72-result.html ../src/harnesses/saxon/saxon-xslt-harness.xproc
-	run java -cp ${SAXON_CP} net.sf.saxon.Query -s:xspec/xspec-72-result.html -qs:"declare default element namespace 'http://www.w3.org/1999/xhtml'; concat(/html/head/meta[@http-equiv eq 'Content-Type']/@content = 'text/html; charset=UTF-8', '&#x0A;')" !method=text
-	echo $output
+    if [ -z ${XMLCALABASH_CP} ]; then
+		    skip "test for XProc skipped as XMLCalabash uses a higher version of Saxon";
+	  fi 
+
+	  run java -Xmx1024m -cp ${XMLCALABASH_CP} com.xmlcalabash.drivers.Main -isource=xspec-72.xspec xspec-home=file://${TRAVIS_BUILD_DIR}/ -oresult=${TRAVIS_BUILD_DIR}/test/xspec/xspec-72-result.html ${TRAVIS_BUILD_DIR}/src/harnesses/saxon/saxon-xslt-harness.xproc 
+	  run java -cp ${SAXON_CP} net.sf.saxon.Query -s:${TRAVIS_BUILD_DIR}/test/xspec/xspec-72-result.html -qs:"declare default element namespace 'http://www.w3.org/1999/xhtml'; /html/head/meta[@http-equiv eq 'Content-Type']/@content = 'text/html; charset=UTF-8'" !method=text
```
Comment (AirQuick) [inline test/xspec.bats]: Can you take the `run java` lines from #86? It corresponds to the initial comment of #86.
> * `xspec.bats` is also modified a bit. (in order to be more platform independent)
> * `'&#x0A;'` is added to XQuery. (in order to avoid conflict with the "Picked up JAVA_TOOL_OPTIONS..." stderr line which is emitted when the `JAVA_TOOL_OPTIONS` environment variable is defined)

Comment (AirQuick) [inline test/xspec.bats]: Oops. Already merged to master. I'll make another pull request for this...

Comment (cirulls) [inline test/xspec.bats]: Sorry, for once I was too quick ;-) Yes, please make another pull request so that we can trace every change. Thanks.

Comment (AirQuick) [inline test/xspec.bats]: submitted #98
